core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java

```diff
@@ -188,7 +190,8 @@ public static void run(Application application, Class<? extends QuarkusApplicati
                         applicationLogger.warn("You can try to kill it with 'kill -9 <pid>'.");
                     }
                 } else if (rootCause instanceof ConfigurationException) {
-                    System.err.println(rootCause.getMessage());
+                    applicationLogger.errorv(e, rootCause.getMessage());
+                    ensureConsoleLogsDrained();
```

Comment:
I think the point of using `System.err` was that the configuration exception might have occurred before setting up the loggers... in which case you don't want to use the logger to print the exception :)

You should probably avoid this if you can, or fix it in a separate commit or PR if you think it's wrong for some reason: that's (hopefully) not related to #36376

Reply:
Changed backed to ```System.err``` I also added a line to print stack trace to achieve the same result since it might be beneficial to user :)